src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/QuestionRepositoryAsyncIT.java

```diff
@@ -49,58 +49,50 @@ public void testSaveAsync() {
         this.repository.deleteAll();
         final Question question = new Question("id", "link");
 
-        this.repository.saveAsync(question).subscribe(a -> {
-            Assert.assertEquals(a, question);
-            Assert.assertTrue(this.repository.findById(question.getId()).isPresent());
-            Assert.assertEquals(this.repository.findById(question.getId()).get(), question);
-        });
+        Question found = this.repository.saveAsync(question).toBlocking().single();
+        Assert.assertEquals(found, question);
 
         question.setUrl("new-link");
-
-        this.repository.saveAsync(question).subscribe(a -> {
-            Assert.assertEquals(a, question);
-            Assert.assertTrue(this.repository.findById(question.getId()).isPresent());
-            Assert.assertEquals(this.repository.findById(question.getId()).get(), question);
-        });
+        found = this.repository.saveAsync(question).toBlocking().single();
+        Assert.assertEquals(found, question);
     }
 
     @Test
     public void testFindByIdAsync() {
-        this.repository.findByIdAsync(QUESTION.getId()).subscribe(a -> Assert.assertEquals(a, QUESTION));
-        this.repository.findByIdAsync(NOT_EXIST_ID).subscribe(
-                a -> {
-                    Assert.assertTrue(false); // should not reach here.
-                },
-                e -> {
-                    Assert.assertEquals(e.getClass(), DocumentClientException.class);
-                    Assert.assertEquals(((DocumentClientException) e).getStatusCode(), HttpStatus.SC_NOT_FOUND);
-                }
-        );
+        final Question question = this.repository.findByIdAsync(QUESTION.getId()).toBlocking().single();
+        Assert.assertEquals(question, QUESTION);
+
+        try {
+            this.repository.findByIdAsync(NOT_EXIST_ID).toCompletable().await();
+        } catch (RuntimeException e) {
+            final Throwable cause = e.getCause();
```

Comment:
If it doesn't throw, this case will pass?

Reply:
good catch, defect here, should add assert_not_reached after findByIdAsync